dpp/src/data_contract/data_contract_factory.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::anyhow;
+use serde_json::{Number, Value as JsonValue};
+use std::collections::BTreeMap;
+use std::convert::TryFrom;
+
+use crate::{
+    data_contract::{self, generate_data_contract_id},
+    decode_protocol_entity_factory::DecodeProtocolEntityFactory,
+    errors::{consensus::ConsensusError, ProtocolError},
+    mocks,
+    prelude::Identifier,
+    util::entropy_generator,
+    DashPlatformProtocol,
+};
+
+use super::DataContract;
+
+pub struct DataContractFactory<SR> {
+    dpp: DashPlatformProtocol<SR>,
+    _validate_data_contract: mocks::ValidateDataContract,
+    decode_protocol_entity: DecodeProtocolEntityFactory,
+}
+
+impl<SR> DataContractFactory<SR> {
+    pub fn new(
+        dpp: DashPlatformProtocol<SR>,
+        _validate_data_contract: mocks::ValidateDataContract,
+        decode_protocol_entity: DecodeProtocolEntityFactory,
+    ) -> Self {
+        Self {
+            dpp,
```

Comment:
I probably wouldn't use dpp here and just pass the version from DPP constructor

Reply:
The reason is that later down the road passing dpp to factories will be quite problematic

Reply:
a really good point, fixed